src/IO/ReadHelpers.cpp

```diff
@@ -1955,10 +1982,26 @@ bool tryReadJSONField(String & s, ReadBuffer & buf)
     return readParsedValueInto<bool>(s, buf, parse_func);
 }
 
+template<bool supports_crlf>
+void readTSVFieldImpl(String & s, ReadBuffer & buf)
+{
+    if constexpr (supports_crlf)
+        readEscapedStringIntoImpl<String, false, false>(s, buf);
+    else
+        readEscapedStringIntoImpl<String, false, true>(s, buf);
+}
```

Comment:
Actually, as `readTSVFieldImpl` just calls coresponding `readEscapedStringIntoImpl` we can remove it and just use corresponding `readEscapedStringIntoImpl` in `readTSVField/readTSVFieldCRLF`.